packages/components/src/components/Divider/Divider.tsx

```diff
@@ -2,67 +2,52 @@
 
 import { type ReactElement } from 'react';
 
-export type DividerStyle = 'noline' | 'dashed' | 'solid' | 'dotted';
+import { cn } from '#/utilities/styleUtilities';
 
-export type DividerHeight = 'small' | 'medium' | 'large';
+import { dividerStyles } from './styles/dividerStyles';
+import { lineStyles } from './styles/lineStyles';
+
+type DividerStyle = 'noline' | 'dashed' | 'solid';
+type DividerPadding = 'none' | 'small' | 'medium' | 'large';
+type DividerDirection = 'horizontal' | 'vertical';
+type DividerColor = 'weak' | 'default' | 'strong' | 'x-strong';
 
 export type DividerProps = {
     style?: DividerStyle;
-    height?: DividerHeight;
-    color?: string;
-    vertical?: boolean;
+    padding?: DividerPadding;
+    color?: DividerColor;
+    direction?: DividerDirection;
     'data-test-id'?: string;
-};
-
-const styleMap = {
-    noline: 'tw-border-none',
-    dashed: 'tw-border-dashed',
-    solid: 'tw-border-solid',
-    dotted: 'tw-border-dotted',
-};
-
-const heightMap = {
-    small: 36,
-    medium: 60,
-    large: 96,
+    className?: string;
 };
 
 const DIVIDER_TEST_ID = 'fondue-divider';
+const DIVIDER_LINE_TEST_ID_SUFFIX = '-line';
 
 export const Divider = ({
-    vertical = false,
-    style = 'solid',
-    height = 'small',
     'data-test-id': dataTestId = DIVIDER_TEST_ID,
-    color = '#CCC',
+    direction = 'horizontal',
+    className,
+    ...props
 }: DividerProps): ReactElement => {
-    return vertical ? (
-        <div
-            aria-hidden="true"
-            className="tw-flex tw-self-stretch tw-mt-0 tw-mb-0 tw-items-center tw-justify-center"
-            data-test-id={dataTestId}
-            style={{
-                marginLeft: heightMap[height] / 2,
-                marginRight: heightMap[height] / 2,
-            }}
-        >
-            <div
-                className={`tw-w-px tw-h-full tw-border-r tw-m-0 ${styleMap[style]}`}
-                style={{ borderRightColor: color }}
-                data-test-id="fondue-divider-line"
-            />
-        </div>
-    ) : (
+    return (
         <div
```

Comment:
no way to removee the wrapping div by setting the `hr` width 1px and height 100%?